libs/wire-subsystems/test/unit/Wire/UserGroupSubsystem/InterpreterSpec.hs

```diff
@@ -412,9 +412,11 @@ spec = timeoutHook $ describe "UserGroupSubsystem.Interpreter" do
                           pageSize = PageSize $ unsafeRange pageSize',
                           sortOrder = sortOrder'
                         }
-                  if length p.page < fromIntegral pageSize'
-                    then pure [p]
-                    else (p :) <$> go (offset + fromJust (mkPosInt32 pageSize'))
+                  let len = length p.page
+                  if
+                    | len > 0 && len < fromIntegral pageSize' -> pure [p]
+                    | len == 0 -> pure []
```

Comment:
ah, so the trick is to fetch the empty page, but just not return it?

in that case i liked the previous test better.  :-)

it's slowly clearing up in my mind now: sometimes pagination has a dedicated hasMore flag that allows us to skip the fetching of the empty page, but we don't have that here and that's fine, too.